src/EllCrv/Finite.jl

```diff
@@ -807,9 +807,12 @@ function trace_of_frobenius(E::EllCrv{T}, n::Int) where T<:FinFieldElem
   a = q +1 - order(E)
   R, x = PolynomialRing(QQ)
   f = x^2 - a*x + q
-  L, alpha = NumberField(f)
-  rest = numerator(alpha^n + (-alpha + a)^n)
-  return q^n + 1 - rest
+  if isirreducible(f)
+    L, alpha = NumberField(f)
+  else
+    alpha = roots(f)[1]
```

Comment:
@JHanselman: Do you have an example where it uses the `else` clause?

Reply:
E = elliptic_curve_from_j_invariant(GF(2,4)(0))
trace_of_frobenius(E,2) == 32
trace_of_frobenius(E,3) == 128

should do it.